frontend/src/app/core/exercise.service.ts

```diff
@@ -60,6 +61,9 @@ export class ExerciseService {
         SocketResponse
     >;
 
+    private readonly proposedActions: Immutable<ExerciseAction>[] = [];
+    private readonly performedActions: Immutable<ExerciseAction>[] = [];
```

Comment:
I do not konw why do you create those here... They are never read or otherwise used according to VSCode. And I don't feel like that we should have them in that spot just for testing

Reply:
I forgot to remove them in a refactoring, thanks ^^

Reply:
See 450f945dd0c502eaf53a51e87be8e27b05509508